code/datums/beam.dm

```diff
@@ -28,14 +28,18 @@
 	var/beam_type = /obj/effect/ebeam
 	///This is used as the visual_contents of beams, so you can apply one effect to this and the whole beam will look like that. never gets deleted on redrawing.
 	var/obj/effect/ebeam/visuals
+	///The color of the beam we're drawing.
+	var/beam_color
 
-/datum/beam/New(beam_origin,beam_target,beam_icon='icons/effects/beam.dmi',beam_icon_state="b_beam",time=INFINITY,maxdistance=INFINITY,btype = /obj/effect/ebeam)
+/datum/beam/New(beam_origin,beam_target,beam_icon='icons/effects/beam.dmi',beam_icon_state="b_beam",time=INFINITY,maxdistance=INFINITY,btype = /obj/effect/ebeam, _beam_color = null)
 	origin = beam_origin
 	target = beam_target
 	max_distance = maxdistance
 	icon = beam_icon
 	icon_state = beam_icon_state
 	beam_type = btype
+	if(_beam_color)
+		beam_color = _beam_color
```

Comment:
don't use _varname, instead just set `src.beam_color = beam_color`. If nothing is sent in as an arg, it will set null as null. Otherwise the var will get set all well and good.